### PR TITLE
Improve the push docker images workflow

### PR DESCRIPTION
**Pull request by YisDav**
### Description

This PR merges several changes to the `Procedure_push_docker_images.yml` workflow file that allows a smoother execution.
Further details can be found in the related issue: https://github.com/wazuh/wazuh-automation/issues/1864

### CODE REVIEW

File: .github/workflows/Procedure_push_docker_images.yml [hunk 1]
```diff
@@ -11,23 +11,62 @@ on:
...
         description: 'Package revision'
         default: '1'
         required: true
+        type: string
       PUSH_IMAGES:
         description: 'Push images'
         type: boolean
         default: true
         required: true
+      ID:
```
Comment (teddytpc1) [inline .github/workflows/Procedure_push_docker_images.yml:66]: The `ID` input should be lowercase: `id`.

File: .github/workflows/Procedure_push_docker_images.yml [hunk 2]
```diff
@@ -109,6 +149,6 @@ jobs:
         IFS=',' read -r -a images <<< "$IMAGE_NAMES"
         for image in "${images[@]}"; do
           echo "Tagging and pushing wazuh/$image:${WAZUH_VERSION} to wazuh/$image:$IMAGE_TAG"
-          docker tag wazuh/$image:${WAZUH_VERSION} wazuh/$image:$IMAGE_TAG
-          docker push wazuh/$image:$IMAGE_TAG
+          docker tag wazuh/$image:${WAZUH_VERSION} ${{ secrets.DOCKERHUB_USERNAME }}/$image:$IMAGE_TAG
```
Comment (teddytpc1) [inline .github/workflows/Procedure_push_docker_images.yml:152]: Please, restore these changes: replace ${{ secrets.DOCKERHUB_USERNAME }} by `wazuh` in lines 152 and 153.

File: .github/workflows/Procedure_push_docker_images.yml [hunk 3]
```diff
@@ -11,23 +11,62 @@ on:
...
+      REVISION:
+        description: 'Package revision'
+        default: '1'
+        required: true
+      PUSH_IMAGES:
+        description: 'Push images'
+        type: boolean
+        default: true
+        required: true
+      ID:
```
Comment (teddytpc1) [inline .github/workflows/Procedure_push_docker_images.yml:31]: The ID input should be lowercase: id.

**Review by teddytpc1 — CHANGES_REQUESTED** [review]

File: .github/workflows/Procedure_push_docker_images.yml (hunk 1 above)
Comment (YisDav) [reply .github/workflows/Procedure_push_docker_images.yml]: Done ✅

File: .github/workflows/Procedure_push_docker_images.yml (hunk 2 above)
Comment (YisDav) [reply .github/workflows/Procedure_push_docker_images.yml]: Done ✅

File: .github/workflows/Procedure_push_docker_images.yml (hunk 3 above)
Comment (YisDav) [reply .github/workflows/Procedure_push_docker_images.yml]: Done ✅

**Review by teddytpc1 — APPROVED** [review]

File: .github/workflows/Procedure_push_docker_images.yml [hunk 4]
```diff
@@ -1,33 +1,84 @@
...
         required: true
-      PRODUCTS:
+      id:
+        description: "ID used to identify the workflow uniquely."
+        type: string
+        required: false
+      dev:
+        description: "Add tag suffix '-dev' to the image tag ?"
+        type: boolean
+        default: false
```
Comment (teddytpc1) [inline .github/workflows/Procedure_push_docker_images.yml:39]: Change the `dev` default value to true for the workflow_dispatch.

**Review by teddytpc1 — CHANGES_REQUESTED** [review]

File: .github/workflows/Procedure_push_docker_images.yml (hunk 4 above)
Comment (YisDav) [reply .github/workflows/Procedure_push_docker_images.yml]: Done ✅

**Review by teddytpc1 — APPROVED** [review]

**Pull request merged**